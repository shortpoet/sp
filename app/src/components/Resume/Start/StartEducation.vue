<template>
  <section class="resume-section p-3 px-lg-5 d-flex flex-column" id="education">
    <div class="my-auto">
      <h2 class="heading heading-2 mb-4">Education</h2>
      <div class="resume-item d-flex flex-column flex-md-row mb-4" v-for="(ed, i) in educations" :key="i">
        <div class="resume-content mr-auto">
          <h3 class="heading heading-3 mb-0">{{ ed.institution }}</h3>
          <div class="start-degree subheading mb-3">{{ ed.degree }}</div>
          <div class="start-education-details">{{ ed.details }}</div>
          <!-- <p>GPA: 3.23</p> -->
        </div>
        <div class="resume-date text-md-right mt-1">
          <span class="text-primary">{{ ed.focus }}</span>
        </div>
      </div>
    </div>
  </section>
</template>

<script>
export default {
  name: 'StartEducation',
  props: {
    educations: {
      type: Array
    }
  }
}
</script>
