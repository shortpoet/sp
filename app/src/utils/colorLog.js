const chalk = require('chalk')
import {inspect} from 'util'

// const _log  = () => Function.prototype.bind.call(console.log, console, 'LOG_ALIAS')

export const log = (color, message) => console.log(chalk[`${color}`](`${inspect(message)}`))

export const colorLog = (message, color, background) => {
  color = color || "black";
  if (background) {
    console.log(`%c` + `${message}`, `color:` + `${color};background:${background}`)
  } else {
    console.log(`%c` + `${message}`, `color:` + `${color}`)
  }
}

// module.exports =  {
//   log,
//   colorLog
// }
