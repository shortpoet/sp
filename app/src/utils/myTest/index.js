import createWrapper from './test.utils.createWrapper'
import factory from './test.utils.factory'
import factoryShallow from './test.utils.factoryShallow'
import propFinder from './test.utils.propFinder'
import propsFinder from './test.utils.propsFinder'
import propMocker from './test.utils.propMocker'
import propsMocker from './test.utils.propsMocker'
import textMatcher from './test.utils.textMatcher'
import textMatcherFactory from './test.utils.textMatcherFactory'

export default {
  createWrapper,
  factory,
  factoryShallow,
  propFinder,
  propsFinder,
  propMocker,
  propsMocker,
  textMatcher,
  textMatcherFactory
}
