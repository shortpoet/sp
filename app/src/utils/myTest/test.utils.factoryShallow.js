import { shallowMount } from '@vue/test-utils'

export const factoryShallow = (
  component, 
  /* istanbul ignore next */
  options = {
    propsData: {},
    localVue: {},
    mocks: {},
    store: {},
    // context: {}, // only for functional components
    router: {},
    computed: {},
    stubs: {},
    slots: {},
    scopedSlots: {},
    attrs: {},
    attachToDocument: false,
    attachTo: null, //HTML Element or String
    listeners: {},
    parentComponent: {},
    provide: {}
  }) => {
    return shallowMount(component, options)
  }

module.exports = factoryShallow