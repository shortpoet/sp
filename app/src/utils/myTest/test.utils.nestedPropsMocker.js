

// // this is incomplete
// // maybe try and develop using TDD  
  
// export const nestedPropsMocker = (props) => {

//   console.log(props)

//   let propsData = {}


//   const handleArray = (props) => {
//     propsData = {}
//     props.map(prop => {
//       propsData[`${prop}`] = `sample ${prop}`
//     })
//     return {
//       propsData
//     }
//   }


//   const handleNested = (props, propsData, previousKey, nextKey) => {
//     console.log(props, propsData)

//     let firstLevel
//     if (previousKey === null && nextKey === null) {
//       firstLevel = true
//     }

//     // spread to clone so the splice works on subsequent tests
//     let values = [...props]

//     const length = parseInt(values.splice(-1, 1)[0])
//     console.log(length)
//     const iterator = [...Array(length).keys()]

//     if (firstLevel === true) {
//       previousKey = values.splice(0, 1)[0]
//       propsData[`${previousKey}`] = []
//     } else {
//       console.log('#### not first run ####')
//       console.log(propsData)
//       console.log(previousKey)
//       nextKey = values.splice(0, 1)[0]
//       console.log(nextKey)
//       propsData[`${previousKey}`][`${nextKey}`] = []
//     }


//     let keyDict

//     for (let i of iterator) {
//       keyDict = {}
//       values.map(prop => {
//         if (Array.isArray(prop)) {
//           console.log(prop)
//           console.log('#### calling handleNested from nested ####')
//           // this would only work for one level nested
//           // TODO maybe refactor for infinite recursion
//           console.log(previousKey)
//           console.log(nextKey)
//           console.log(propsData)
//           handleNested(prop, propsData, previousKey, nextKey)
//         } else {
//           console.log(prop)
//           keyDict[`${prop}`] = `sample ${prop}`
//         }
//       })
//       if (firstLevel === true) {
//         propsData[`${previousKey}`].push(keyDict)
//         firstRun = false
//       } else {
//         console.log(propsData)
//         console.log(previousKey)
//         console.log(nextKey)
//         console.log(propsData.experiences)
//         console.log(propsData.experiences.jobs)
//         propsData[`${previousKey}`][`${nextKey}`].push(keyDict)
//       }
//     }
//     return {
//       propsData
//     }
//   }

//   const isNested = props.some(p => Array.isArray(p))
//   const previousKey = null
//   const nextKey = null

//   if (isNested) {
//     console.log('#### is nested ####')
//     handleNested(props, propsData, previousKey, nextKey)
//   } else  {
//     handleArray(props)
//   }
// }
