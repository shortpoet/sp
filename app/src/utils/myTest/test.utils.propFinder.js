import {hardResume} from '@/assets/resume.js'

export const propFinder = (prop) => {
  return {
    propsData: {
      [`${prop}`]: hardResume[`${prop}`]
    }
  }
}

module.exports = propFinder