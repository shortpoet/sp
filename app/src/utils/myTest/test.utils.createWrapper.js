import { createLocalVue } from '@vue/test-utils'
import factory from './test.utils.factory'
import factoryShallow from './test.utils.factoryShallow'

// import VueRouter from 'vue-router'
import Vuex from 'vuex'
import PortalVue from 'portal-vue'

import createStore from '@/store/createStore'
// import createMockRouterConfig from '@/router/createMockRouterConfig'
// import paths from '@/router/paths.js'

import { log } from '@/utils/colorLog'
export const createWrapper  = (
  component,
  /* istanbul ignore next */
  options = {},
  // but why don't i have to ignore the empty  or even unprovided resumeStoreOtions
  // might need to figure this out but setting params to ignore for now
  resumeStoreOptions = {},
  // routerOptions = {},
  isShallow = false,
  ) => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(PortalVue)
    const mockStoreResume = createStore.createMocks().createStoreResumeMocks(resumeStoreOptions)
    const storeConfig = createStore.createMocks({modules: {resume: mockStoreResume}})

    // Using router is untested makes tests fail
    // and just doesn't work for now
    // localVue.use(VueRouter)
    // const routerConfig = createMockRouterConfig.createMockRouterConfig(routerOptions)
    // const router = new VueRouter(routerConfig)

    // log('green', mockStoreResume)
    // log('yellow', storeConfig.modules.resume)
    const store = new Vuex.Store(storeConfig)
    // log('red', store)
    // log('blue', store._mutations['resume/SET_RESUME_RAW'][0]())
    // console.log(options)
    return isShallow ? 
    factoryShallow(component, {
      localVue,
      store,
      // router,
      ...options
    }) : 
    factory(component, {
      localVue,
      store,
      // router,
      ...options
    })
}

// export default createWrapper
// when this was set to export default a console log(createWrapper) that was in
// test.utils.textMatcher line 19 was being executed from every module that called createWrapper
module.exports = createWrapper