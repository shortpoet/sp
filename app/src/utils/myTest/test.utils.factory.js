import { mount } from '@vue/test-utils'
import { log } from '@/utils/colorLog'
export const factory = (
  component, 
  /* istanbul ignore next */
  options = {
    propsData: {},
    localVue: {},
    mocks: {},
    store: {},
    // context: {}, // only for functional components
    // this only breaks when testing
    // probably should comment out of real func (which this is (now) hehe)
    router: {},
    computed: {},
    stubs: {},
    slots: {},
    scopedSlots: {},
    attrs: {},
    attachToDocument: false,
    attachTo: null, //HTML Element or String
    listeners: {},
    parentComponent: {},
    provide: {}
  }) => {
    // console.log(options)
    // console.log(options.mocks)
    // console.log(options.store)
    // console.log(options.localVue)
    // log('blue', options.stubs)
    // log('green', options.propsData)
    // console.log(mount)
    return mount(component, options)
  }

module.exports = factory

// export default factory