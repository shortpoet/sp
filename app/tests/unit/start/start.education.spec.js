import * as myTest from '@/utils/myTest'
const { textMatcher, propsMocker, createWrapper } = myTest.default
import StartEducation from '@/components/Resume/Start/StartEducation'

describe('StartEducation.vue', () => {

  const component = StartEducation
  // const props = Object.keys(component.props)
  const props = {
    educations: ['institution', 'degree', 'details', 'focus', '3']
  }  

  const propsObject = propsMocker(props)
  
  const wrapper = createWrapper(component, propsMocker(props))

  let prop
  let selector
  let mockProp = true

  beforeEach(() => {
    jest.resetModules()
    jest.clearAllMocks()
  })

  it('renders education h2', () => {

    // hard coded header
    expect(wrapper.find("h2").text()).toMatch("Education")

  })
  it('renders education h3 that matches institution prop', () => {

    prop = 'institution'
    selector = 'h3'
    textMatcher(component, props, prop, selector)

  })
  it('renders education .start-degree that matches degree prop', () => {

    prop = 'degree'
    selector = '.start-degree'
    textMatcher(component, props, prop, selector)

  })
  it('renders education .start-education-details that matches details prop', () => {

    prop = 'details'
    selector = '.start-education-details'
    textMatcher(component, props, prop, selector)

  })
  it('renders education span that matches focus prop', () => {

    prop = 'focus'
    selector = 'span'
    textMatcher(component, props, prop, selector)

  })
  it('matches education snapshot', () => {
    
    expect(wrapper.html()).toMatchSnapshot()

  })
})
