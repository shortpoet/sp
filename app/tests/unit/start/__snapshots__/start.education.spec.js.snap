// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartEducation.vue matches education snapshot 1`] = `
<section id="education" class="resume-section p-3 px-lg-5 d-flex flex-column">
  <div class="my-auto">
    <h2 class="heading heading-2 mb-4">Education</h2>
    <div class="resume-item d-flex flex-column flex-md-row mb-4">
      <div class="resume-content mr-auto">
        <h3 class="heading heading-3 mb-0">sample institution</h3>
        <div class="start-degree subheading mb-3">sample degree</div>
        <div class="start-education-details">sample details</div>
      </div>
      <div class="resume-date text-md-right mt-1"><span class="text-primary">sample focus</span></div>
    </div>
    <div class="resume-item d-flex flex-column flex-md-row mb-4">
      <div class="resume-content mr-auto">
        <h3 class="heading heading-3 mb-0">sample institution</h3>
        <div class="start-degree subheading mb-3">sample degree</div>
        <div class="start-education-details">sample details</div>
      </div>
      <div class="resume-date text-md-right mt-1"><span class="text-primary">sample focus</span></div>
    </div>
    <div class="resume-item d-flex flex-column flex-md-row mb-4">
      <div class="resume-content mr-auto">
        <h3 class="heading heading-3 mb-0">sample institution</h3>
        <div class="start-degree subheading mb-3">sample degree</div>
        <div class="start-education-details">sample details</div>
      </div>
      <div class="resume-date text-md-right mt-1"><span class="text-primary">sample focus</span></div>
    </div>
  </div>
</section>
`;
